mozilla_django_oidc_db/config.py

```diff
@@ -110,51 +124,53 @@ def store_config(request: HttpRequest) -> None:
-    config_class = ""
+    # The config_identifier key is added to the state in the OIDCAuthenticationRequestInitView.get method.
+    # The state parameter is present in the error flow if it was present in the Authorization request: https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+    config_identifier = ""
     state_key = request.GET.get("state")
     if state_key and state_key in (states := request.session.get("oidc_states", [])):
         state = states[state_key]
-        config_class = state.get("config_class", "")
+        config_identifier = state.get("config_identifier", "")
 
-    if not config_class and (
-        _config := request.session.get(CONFIG_CLASS_SESSION_KEY, "")
+    if not config_identifier and (
+        _config := request.session.get(CONFIG_IDENTIFIER_SESSION_KEY, "")
     ):
-        config_class = _config
+        config_identifier = _config
 
     try:
-        config = apps.get_model(config_class)
-    except (LookupError, ValueError) as exc:
-        raise BadRequest("Could not look up the referenced config.") from exc
+        config = OIDCConfig.objects.select_related("oidc_provider_config").get(
+            identifier=config_identifier
+        )
+    except (ObjectDoesNotExist, MultipleObjectsReturned) as exc:
+        raise BadRequest("Could not look up the referenced configuration.") from exc
 
     # Spoofing is not possible since we store it in the server-side session, but there
     # can still be all sorts of programmer mistakes.
-    if not issubclass(config, OpenIDConnectConfigBase):
+    if not isinstance(config, OIDCConfig):
         raise BadRequest("Invalid config referenced.")
 
-    request._oidcdb_config_class = config  # type: ignore
+    request._oidcdb_config = config
 
 
-def lookup_config(request: HttpRequest) -> type[OpenIDConnectConfigBase]:
+def lookup_config(request: HttpRequest) -> OIDCConfig:
     # cache on request for optimized access -- preferred access
-    if config := getattr(request, "_oidcdb_config_class", None):
+    if config := getattr(request, "_oidcdb_config", None):
         return config
 
     # if not cached, try to reconstruct from session
     if (session := getattr(request, "session", None)) is None:
         raise BadRequest("No session present on request")
 
-    if (config_class := session.get(CONFIG_CLASS_SESSION_KEY)) is None:
+    if (config_identifier := session.get(CONFIG_IDENTIFIER_SESSION_KEY)) is None:
         raise BadRequest("The required config is not available on the session.")
 
     try:
-        config = apps.get_model(config_class)
-    except (LookupError, ValueError) as exc:
-        raise BadRequest("Could not look up the referenced config.") from exc
+        config = OIDCConfig.objects.select_related("oidc_provider_config").get(
+            identifier=config_identifier
+        )
+    except (ObjectDoesNotExist, MultipleObjectsReturned) as exc:
+        raise BadRequest("Could not look up the referenced configuration.") from exc
```

Comment:
this pattern repeats, perhaps it's a good idea to encapsulate it in a custom manager method:

```py
config = OIDCConfig.objects.resolve(identifier)
```